### DISCUSSION TITLE
Table Column Filter: confirm on reset

### DISCUSSION THREAD

**Discussion by Olivolution** [ant-design/ant-design#51053] [Ideas]
Hello antd friends 😃 

Maybe I've overlooked a simple way to do this, but when resetting my filters in a table column filter, I'd like the reset to apply immediately. The current behaviour expects an additional click on "confirm" before the reset takes hold.
![grafik](https://github.com/user-attachments/assets/94f26f1b-1681-4781-879a-cb84cfde2da3)

That behaviour is intuitive when it comes to adding filters, but many users expect a reset to happen without an additional click. So far the only solution I found to this was to completely overwrite and rebuild my own filterDropdown within the available prop, which seems like quite the overkill for my goal.

**Would it be possible to get an "onReset" prop that takes a function or a "confirmOnReset" boolean prop?**
This whole idea is referring to this section of your docs: https://ant.design/components/table#column

Thank you and kind regards 👋 

Edit: I realized the confirm() function is not easily accessible outside of the filterDropdown. In hindsight I think a simple boolean prop to toggle confirmOnFilterReset true/false (default: false), would make for a simpler solution here. If a full onReset prop were to be implemented, the confirm() function should also somehow be destructurable from the antd Table, which I can imagine to be a more tedious change to make

**Comment by petercat-assistant** [comment]
It appears that there are no existing issues or pull requests directly addressing the feature request for an "onReset" prop or a "confirmOnReset" boolean prop for the Table Column Filter in Ant Design.

Your request seems to be a new feature. I recommend waiting for feedback from the maintainers or contributors of the Ant Design project.

For further assistance, please describe your question in the comments and @petercat-assistant to start a conversation with me.
